src/dawn/CodeGen/CXXNaive/CXXNaiveCodeGen.cpp

```diff
@@ -0,0 +1,527 @@
+    ctrArgs += "," + SIRFieldsWithoutTemps[i]->Name;
+  }
+
+  // add the ctr initialization of each stencil
+  for(std::size_t stencilIdx = 0; stencilIdx < stencils.size(); ++stencilIdx) {
+
+    const Stencil& stencil = *stencilInstantiation->getStencils()[stencilIdx];
+
+    if(stencil.isEmpty())
+      continue;
+
+    const auto& StencilFields = stencil.getFields();
+
+    std::string initCtr =
+        "m_" + innerStencilNames[stencilIdx] + "(new " + innerStencilNames[stencilIdx];
+
+    int i = 0;
+    for(auto field : StencilFields) {
+      if(field.IsTemporary)
+        continue;
+      initCtr +=
+          (i != 0 ? "," : "<") + (stencilInstantiation->isAllocatedField(field.AccessID)
+                                      ? (c_gtc().str() + "storage_t")
+                                      : (std::string("StorageType") + std::to_string(i + 1)));
+      i++;
+    }
+    initCtr += ">(dom";
+    for(auto field : StencilFields) {
+      if(field.IsTemporary)
+        continue;
+      initCtr += "," + (stencilInstantiation->isAllocatedField(field.AccessID) ? (field.Name + "_")
+                                                                               : (field.Name));
+    }
+    initCtr += ") )";
+    StencilWrapperConstructor.addInit(initCtr);
+  }
+
+  if(stencilInstantiation->hasAllocatedFields()) {
+    StencilWrapperConstructor.addInit("m_meta_data(dom.isize(), dom.jsize(), dom.ksize())");
+    for(int AccessID : stencilInstantiation->getAllocatedFieldAccessIDs())
+      StencilWrapperConstructor.addInit(
+          stencilInstantiation->getNameFromAccessID(AccessID) + "_(m_meta_data,\"" +
+          stencilInstantiation->getNameFromAccessID(AccessID) + "\")");
+  }
+
+  StencilWrapperConstructor.commit();
+
+  // Generate the run method by generate code for the stencil description AST
+  MemberFunction RunMethod = StencilWrapperClass.addMemberFunction("void", "run", "");
+
+  RunMethod.finishArgs();
+  // Create the StencilID -> stencil name map
+  std::unordered_map<int, std::vector<std::string>> stencilIDToStencilNameMap;
+  for(std::size_t i = 0; i < stencils.size(); ++i)
+    stencilIDToStencilNameMap[stencils[i]->getStencilID()].emplace_back(innerStencilNames[i]);
+
+  // generate the control flow code executing each inner stencil
+  ASTStencilDesc stencilDescCGVisitor(stencilInstantiation, stencilIDToStencilNameMap);
+  stencilDescCGVisitor.setIndent(RunMethod.getIndent());
+  for(const auto& statement : stencilInstantiation->getStencilDescStatements()) {
+    statement->ASTStmt->accept(stencilDescCGVisitor);
+    RunMethod.addStatement(stencilDescCGVisitor.getCodeAndResetStream());
+  }
+
+  RunMethod.commit();
+
+  StencilWrapperClass.commit();
+
+  // Remove trailing ';' as this is retained by Clang's Rewriter
+  std::string str = ssSW.str();
+  str[str.size() - 2] = ' ';
+
+  return str;
+}
+
+std::string CXXNaiveCodeGen::generateGlobals(const SIR* Sir) {
```

Comment:
why is the variable name starting with an uppercase letter?

Reply:
this one is coming from fabian, sometimes we are not following the conventions. I changed it.